M2/Macaulay2/packages/MatrixSchubert/permutationMethods.m2

```diff
@@ -226,10 +226,14 @@ isPatternAvoiding (List,List) := Boolean => (perm, pattern) -> (
     --assume permutation is pattern-avoiding, break if not true
     isAvoiding := true;
     for idx in subsets(0..#perm-1, #pattern) do {
```

Comment:
This isn't really a comment for this PR specifically, but just a comment in general regarding `subsets`.  It would be great if it returned at iterator (maybe as an option) so we wouldn't need to pre-compute *all* $\binom{n}{k}$ subsets at the beginning.  If we find out that it's not pattern avoiding early in the process, then we would have computed all the other subsets for no reason.